src/NzbDrone.Core/MediaFiles/MediaInfo/UpdateMediaInfoService.cs

```diff
@@ -55,18 +55,18 @@ public void Handle(SeriesScannedEvent message)
             }
         }
 
-        public bool Update(EpisodeFile episodeFile, Series series)
+        public bool Update(EpisodeFile episodeFile, Series series, bool update = true)
         {
             if (!_configService.EnableMediaInfo)
             {
                 _logger.Debug("MediaInfo is disabled");
                 return false;
             }
 
-            return UpdateMediaInfo(episodeFile, series);
+            return UpdateMediaInfo(episodeFile, series, update);
         }
 
-        private bool UpdateMediaInfo(EpisodeFile episodeFile, Series series)
+        private bool UpdateMediaInfo(EpisodeFile episodeFile, Series series, bool update = true)
```

Comment:
If the mediainfo for a file needs to be reprocessed before the file in in the DB that shouldn't be done with this, which is meant to update media info for files already stored in the DB.

Reply:
I guess I can also just break it into two methods, one which updates the `MediaInfo` and one which does that plus writes to the DB.

Reply:
Just getting the media info for the updated file can be retrieved directly from `_videoFileInfoReader.GetMediaInfo` and then set on the episode file.